src/streams.rs

```diff
@@ -0,0 +1,698 @@
+    pub idle: usize,
+}
+
+/// A group parsed from [`xinfo_groups`] command.
+///
+/// [`xinfo_groups`]: ../trait.Commands.html#method.xinfo_groups
+///
+#[derive(Default, Debug, Clone)]
+pub struct StreamInfoGroup {
+    /// The group name.
+    pub name: String,
+    /// Number of consumers known in the group.
+    pub consumers: usize,
+    /// Number of pending messages (delivered but not yet acknowledged) in the group.
+    pub pending: usize,
+    /// Last ID delivered to this group.
+    pub last_delivered_id: String,
+}
+
+/// Represents a pending message parsed from [`xpending`] methods.
+///
+/// [`xpending`]: ../trait.Commands.html#method.xpending
+#[derive(Default, Debug, Clone)]
+pub struct StreamPendingId {
+    /// The ID of the message.
+    pub id: String,
+    /// The name of the consumer that fetched the message and has
+    /// still to acknowledge it. We call it the current owner
+    /// of the message.
+    pub consumer: String,
+    /// The number of milliseconds that elapsed since the
+    /// last time this message was delivered to this consumer.
+    pub last_delivered_ms: usize,
+    /// The number of times this message was delivered.
+    pub times_delivered: usize,
+}
+
+/// Represents a stream `key` and its `id`'s parsed from `xread` methods.
+#[derive(Default, Debug, Clone)]
+pub struct StreamKey {
+    /// The stream `key`.
+    pub key: String,
+    /// The parsed stream `id`'s.
+    pub ids: Vec<StreamId>,
+}
+
+impl StreamKey {
+    /// Return only the stream `id`'s without their data.
+    pub fn just_ids(&self) -> Vec<&String> {
+        self.ids.iter().map(|msg| &msg.id).collect::<Vec<&String>>()
+    }
+}
+
+/// Represents a stream `id` and its field/values as a `HashMap`
+#[derive(Default, Debug, Clone)]
+pub struct StreamId {
+    /// The stream `id` (entry ID) of this particular message.
+    pub id: String,
+    /// All fields in this message, associated with their respective values.
+    pub map: HashMap<String, Value>,
+}
+
+impl StreamId {
+    /// Converts a `Value::Bulk` into a `StreamId`.
+    pub fn from_bulk_value(v: &Value) -> RedisResult<Self> {
+        let mut stream_id = StreamId::default();
+        if let Value::Bulk(ref values) = *v {
+            if let Some(v) = values.get(0) {
+                stream_id.id = from_redis_value(&v)?;
+            }
+            if let Some(v) = values.get(1) {
+                stream_id.map = from_redis_value(&v)?;
+            }
+        }
+
+        Ok(stream_id)
+    }
+
+    /// Fetches value of a given field and converts it to the specified
+    /// type.
+    pub fn get<T: FromRedisValue>(&self, key: &str) -> Option<T> {
+        match self.find(&key) {
+            Some(ref x) => from_redis_value(*x).ok(),
+            None => None,
+        }
+    }
+
+    /// Fetches the value of a given field without performing any
+    /// type conversion.
+    pub fn find(&self, key: &&str) -> Option<&Value> {
```

Comment:
Having both `find` and `get` seems confusing/redundant. Users can just call `self.map.get` so perhaps this method can be removed.